nginx-pckg-module/src/media/codec_config.c

```diff
@@ -492,12 +496,38 @@ codec_config_mp4a_config_parse(
 
     if (reader.stream.eof_reached)
     {
-        vod_log_error(VOD_LOG_ERR, log, 0,
-            "codec_config_mp4a_config_parse: failed to read all required audio extra data fields");
-        return VOD_BAD_DATA;
+        goto error;
+    }
+
+    if (result->object_type == AOT_SBR || result->object_type == AOT_PS )
```

Comment:
looking at ffmpeg (ff_mpeg4audio_get_config_gb), I see there is another condition here, I think it should be -
```
temp_reader = reader;
if (result->object_type == AOT_SBR || (result->object_type == AOT_PS &&
    !(bit_read_stream_get(&temp_reader, 3) & 0x03 && !(bit_read_stream_get(&temp_reader, 9) & 0x3f))))
```

Reply:
also style - remove the space before `)`

Reply:
so why didn't u use the ffmpeg function from  the very start?

Reply:
also, u have no bit_show semantics